SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -118,200 +123,72 @@ def evaluate_integral(self, receiver_location, components, tolerance=1e-4):
-                        )
-
-                    arg = dx[:, aa] * dz[:, cc] / dyr
-
-                    if (
-                        ("gyy" in components)
-                        or ("gzz" in components)
-                        or ("guv" in components)
-                    ):
-                        gyy -= (
-                            (-1) ** aa
-                            * (-1) ** bb
-                            * (-1) ** cc
-                            * (
-                                dxdy / (r * dz_r)
-                                + dydz / (r * dx_r)
-                                - np.arctan(arg)
-                                + dy[:, bb]
-                                * (1.0 / (1 + arg ** 2.0))
-                                * dxdz
-                                / dyr ** 2.0
-                                * (r + dy[:, bb] ** 2.0 / r)
-                            )
-                        )
-
-                    if "gyz" in components:
-                        rows["gyz"] -= (
-                            (-1) ** aa
-                            * (-1) ** bb
-                            * (-1) ** cc
-                            * (
-                                np.log(dx_r)
-                                + dz[:, cc] ** 2.0 / (r * (dx_r))
-                                + dx[:, aa] / r
-                                - 1.0
-                                / (1 + arg ** 2.0)
-                                * (dx[:, aa] / (r ** 2))
-                                * (r - dz[:, cc] ** 2.0 / r)
-                            )
-                        )
-
-        if "gyy" in components:
-            rows["gyy"] = gyy
-
-        if "gxx" in components:
-            rows["gxx"] = gxx
-
+        dr = self._nodes - receiver_location
+        dx = dr[..., 0]
+        dy = dr[..., 1]
+        dz = dr[..., 2]
+
+        node_evals = {}
+        if "gx" in components:
+            node_evals["gx"] = prism_fz(dy, dz, dx)
+        if "gy" in components:
+            node_evals["gy"] = prism_fz(dz, dx, dy)
+        if "gz" in components:
+            node_evals["gz"] = prism_fz(dx, dy, dz)
+        if "gxy" in components:
+            node_evals["gxy"] = prism_fzx(dy, dz, dx)
+        if "gxz" in components:
+            node_evals["gxz"] = prism_fzx(dx, dy, dz)
+        if "gyz" in components:
+            node_evals["gyz"] = prism_fzy(dx, dy, dz)
+        if "gxx" in components or "guv" in components:
+            node_evals["gxx"] = prism_fzz(dy, dz, dx)
+        if "gyy" in components or "guv" in components:
+            node_evals["gyy"] = prism_fzz(dz, dx, dy)
+            if "guv" in components:
+                node_evals["guv"] = (node_evals["gyy"] - node_evals["gxx"]) * 0.5
+                # (NN - EE) / 2
+        inside_adjust = False
         if "gzz" in components:
-            rows["gzz"] = -gxx - gyy
-
-        if "guv" in components:
-            rows["guv"] = -0.5 * (gxx - gyy)
```

Comment:
Aren't we losing `guv` here? 

![image](https://user-images.githubusercontent.com/55204635/197596069-3e4ccad7-58e4-4a6c-b0d6-7f3742e0cd3f.png)

Reply:
It is not lost! If you look, there are two conditions to calculate `gxx`, whether `gxx` or `guv` is in components, and the same for `gyy`, then it is finally computed at during the `gyy` check.

Reply:
Yup, my bad.